Parse/Tests/Unit/ParseClientConfigurationTests.m

```diff
@@ -119,8 +119,8 @@ - (void)testExtensionDataSharing {
     ParseClientConfiguration *configuration = [ParseClientConfiguration emptyConfiguration];
 
 #if !PF_TARGET_OS_OSX
-    // Innaccessible bundle identifiers should throw
-    XCTAssertThrows(configuration.applicationGroupIdentifier = @"someBundleIdentifier");
+    // This is nil is no path is set
+    XCTAssertNil(configuration.applicationGroupIdentifier);
```

Comment:
This changes what's being tested.

Following the call stack, it looks like we should throw here if we can't access the container. Maybe I'm rusty, but I can't see why the previous test would ever have thrown an exception? The call at the end of the stack `NSFileManager containerURLForSecurityApplicationGroupIdentifier:` doesn't seem to throw?

Reply:
My guess here was something changed in Objective-C between versions of Xcode. I think this because I tested on Xcode 11 and 12 and checked the debugger. I've seen somethings switch to `nil` and this was one of them. If you run this test in Xcode 11 it will fail. Thinking forward, I switched to the Xcode 12 way

Reply:
I'm not sure what could have been changed to nil, but this change leaves the logic in setApplicationGroupIdentifier: untested:

```
- (void)setApplicationGroupIdentifier:(NSString *)applicationGroupIdentifier {
    PFParameterAssert(applicationGroupIdentifier == nil ||
                      [PFFileManager isApplicationGroupContainerReachableForGroupIdentifier:applicationGroupIdentifier],
                      @"ApplicationGroupContainer is unreachable. Please double check your Xcode project settings.");

    _applicationGroupIdentifier = [applicationGroupIdentifier copy];
}
```

Reply:
You're right. The issue is in Xcode 12, it's letting this be set and isn't throwing. The change I added only tests if the default is nil.

We need to find out why it isn't throwing anymore and allowing `applicationGroupIdentifier` to be set.

Reply:
I think this was meant to test `ParseClientConfiguration.setApplicationGroupIdentifier` and not setting directly. Only this is suppose to use something equivalent to `willSet` in Swift

Reply:
I'm reverting back. I think it's something up with the swizzler for file manager. It's not throwing an error anymore in Xcode 12

Reply:
I remember reading some where that there were changes to how shared group directories work in Xcode 12 which might be related to this test failure

Reply:
IMO, if the rest of the tests pass, I would look for a fix in a future PR. I see two possible issues the test brings up. Either users are able to do something in Xcode 12 (meaning iOS 14) that they weren't able to do in previous versions or there's an access change in PFFileManager that needs to be fixed. In either case, people are already using this SDK in Xcode 12 and this specific PR doesn't need the fix or stop people from using Xcode 12. Running the tests in Xcode 11 would just give us a green check mark, but the issue will still be present.

Reply:
Something that confused me a little here was actually which Xcode version Actions is using. I see macos-latest in the yml, but the release build passes. And release currently fails in Xcode 12 due to the conflicting architecture situation, unless there's something in this changeset that fixes it that I'm missing?

Personally I don't have an issue with letting that test failing test through, but I think the repo rules won't allow us to merge with that error.

Reply:
> Something that confused me a little here was actually which Xcode version Actions is using. I see macos-latest in the yml, but the release build passes. And release currently fails in Xcode 12 due to the conflicting architecture situation, unless there's something in this changeset that fixes it that I'm missing?

Every build is using Xcode 12 latest (default), Release is using 11 because of what you mentioned:
https://github.com/parse-community/Parse-SDK-iOS-OSX/blob/b9bd19a67aa8357cda40934a96b253156ff85f84/.github/workflows/ci.yml#L162-L163

> EDIT: Perhaps we can disable the test and create an issue / PR to fix it, so it doesn't disappear completely.

I agree with this process

Reply:
Ahh yes! I missed that.

Reply:
Shot in the dark, but this might be as simple as a group entitlement issue, https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_application-groups

Reply:
Especially since the way entitlements are used is something that's always changing